torch/_inductor/codegen/cpp.py

```diff
@@ -1299,6 +1275,11 @@ def square(a):
 
     @staticmethod
     def where(a, b, c):
+        assert isinstance(b, CppCSEVariable)
+        if b.dtype != torch.float:
+            raise CppVecUnsupportedError(
+                "where with non-float tensor is not supported in vectorized codegen"
+            )
```

Comment:
I made the codegen to also check the legality of vectorization since the checking in codegen is simpler due to more accurate info.